packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,127 @@
+  late final _userStatusBloc = Provider.of<AccountsBloc>(context, listen: false).getUserStatusesBloc(widget.account);
+
+  @override
+  void initState() {
+    super.initState();
+
+    unawaited(_userStatusBloc.load(widget.username));
+  }
+
+  @override
+  Widget build(final BuildContext context) {
+    final isDark = Theme.of(context).brightness == Brightness.dark;
+    final size = (widget.size * MediaQuery.of(context).devicePixelRatio).toInt();
+    return Stack(
+      alignment: Alignment.center,
+      children: [
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-${widget.username}-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: widget.username,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: widget.username,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        if (widget.showStatus) ...[
+          ResultBuilder<NextcloudUserStatusPublicStatus?>(
+            stream: _userStatusBloc.statuses.map((final statuses) => statuses[widget.username]),
+            builder: _userStatusIconBuilder,
+          ),
+        ],
+      ],
+    );
+  }
+
+  Widget _userStatusIconBuilder(final BuildContext context, final Result<NextcloudUserStatusPublicStatus?> result) {
+    final hasEmoji = result.data?.icon != null;
+    final scaledSize = widget.size / (hasEmoji ? 2 : 3);
+
+    Widget? child;
+    Decoration? decoration;
+    if (result.loading) {
+      child = CircularProgressIndicator(
+        strokeWidth: 1.5,
+        color: Theme.of(context).colorScheme.onPrimary,
+      );
+    } else if (result.error != null) {
+      child = Icon(
+        Icons.error_outline,
+        size: scaledSize,
+        color: Colors.red,
+      );
+    } else if (hasEmoji) {
+      child = Text(
+        result.data!.icon!,
+        style: const TextStyle(
+          fontSize: 16,
+        ),
+      );
+    } else if (result.data != null) {
+      decoration = BoxDecoration(
+        shape: BoxShape.circle,
+        color: _userStatusToColor(result.data!.status),
+      );
+    }
+
+    return SizedBox.square(
+      dimension: widget.size,
+      child: Align(
+        alignment: Alignment.bottomRight,
+        child: SizedBox.square(
+          dimension: scaledSize,
+          child: DecoratedBox(
+            decoration: decoration ?? const BoxDecoration(),
+            child: child,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Color _userStatusToColor(final NextcloudUserStatusType userStatusType) => switch (userStatusType) {
```

Comment:
Can we inline this into the line 102?

Or maybe we'll leave it as is and incorporate it into our `NeonTheme` once we have it?

Reply:
I'm not sure if this will be used somewhere else or not. Maybe for setting the status? Though adding it to the theme doesn't make a lot of sense to me (the color values are pulled from the web UI).

Reply:
I think I would just leave it here as is and move it somewhere else when needed.